include/clink/linalg/vector.h

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2021 The Clink Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef CLINK_LINALG_VECTOR_H_
+#define CLINK_LINALG_VECTOR_H_
+
+#include "tfrt/host_context/chain.h"
+
+namespace clink {
+
+// A vector of double values.
+class Vector {
+public:
+  virtual ~Vector() {}
+
+  // Sets the value at a certain index of the vector.
+  virtual tfrt::AsyncValueRef<tfrt::Chain> set(const int &index,
```

Comment:
Given that set(...) is supposed to return result quickly, ,do we need this method to an `AsyncValueRef`?

Same for get(...).

Reply:
I'll change them to `llvm::Error` and `llvm::Expected`.